src/Client/ClientApplicationBase.cpp

```diff
@@ -171,6 +171,8 @@ void ClientApplicationBase::init(int argc, char ** argv)
 
         ("stage", po::value<std::string>()->default_value("complete"), "Request query processing up to specified stage: complete,fetch_columns,with_mergeable_state,with_mergeable_state_after_aggregation,with_mergeable_state_after_aggregation_and_limit")
         ("progress", po::value<ProgressOption>()->implicit_value(ProgressOption::TTY, "tty")->default_value(ProgressOption::DEFAULT, "default"), "Print progress of queries execution - to TTY: tty|on|1|true|yes; to STDERR non-interactive mode: err; OFF: off|0|false|no; DEFAULT - interactive to TTY, non-interactive is off")
+        ("progress-table", po::value<ProgressOption>()->implicit_value(ProgressOption::TTY, "tty")->default_value(ProgressOption::DEFAULT, "default"), "Print a progress table with changing metrics during query execution - to TTY: tty|on|1|true|yes; to STDERR non-interactive mode: err; OFF: off|0|false|no; DEFAULT - interactive to TTY, non-interactive is off.")
+        ("enable-progress-table-toggle", po::value<bool>()->default_value(true), "Enable toggling of the progress table by pressing the control key (Space). Only applicable in interactive mode with the progress table enabled.")
```

Comment:
I enabled toggling by default, but there is a caveat:
If a client communicates with the server interactively, e.g., using `expect` program, the keystroke interceptor consumes the input and makes the communication impossible. In our `.expect` tests I explicitly disabled the `--enable-progress-table-toggle` option.
Perhaps, we should set this setting to `false` by default to be on the safe side. It may break clients that automate client interaction through `expect`-like programs.

Reply:
I'm totally ok with enabling it by default.
Expect scripts are a rare scenario. The interactive mode is not an API.